doc/querying.md

```diff
@@ -0,0 +1,240 @@
+=> [#:users{:is-admin false, :username test@test.com, :id 1} #:users{:is-admin true, :username admin@test.com, :id 2}]
+
+(select-one! *env* (-> users-rel (r/where [:= 1 :id])))
+=> #:users{:is-admin false, :username test@test.com, :id 1}
+
+(select! *env* :users)
+=> [#:users{:is-admin false, :username test@test.com, :id 1} #:users{:is-admin true, :username admin@test.com, :id 2}]
+
+(select-one! *env* (-> *env* :users (r/where [:= 1 :id])))
+=> #:users{:is-admin false, :username test@test.com, :id 1}
+```
+
+### Specifying a schema
+
+Schema can be specified explicitly through relation spec:
+
+```clojure
+(def users-spec-with-schema {:name "users"
+                 :columns ["id" "username" "is_admin"]
+                 :pk ["id"]
+                 :schema "foo"})
+
+(r/get-select-query (r/spec->relation users-spec-with-schema) {})
+
+=> ["SELECT \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"foo\".\"users\" AS \"users\""]
+```
+
+Schema can also be renamed when necessary:
+
+```clojure
+(require '[com.verybigthings.penkala.env :as env])
+
+(r/get-select-query (r/spec->relation users-spec-with-schema) (env/with-schema-rename "foo" "bar"))
+
+=> ["SELECT \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"bar\".\"users\" AS \"users\""]
+```
+
+## Examples
+
+### where
+
+Adding WHERE clause:
+
+```clojure
+(def admins-rel (r/where users-rel [:is-true :is-admin]))
+
+(r/get-select-query admins-rel {})
+
+=> ["SELECT \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\" WHERE \"users\".\"is_admin\" IS TRUE"]
+```
+
+### distinct
+
+Adding a DISTINCT and DISTINCT ON clause:
+
+```clojure
+(r/get-select-query (r/distinct users-rel) {})
+
+["SELECT DISTINCT \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\""]
+```
+
+```clojure
+(r/get-select-query (r/distinct users-rel [:id]) {})
+
+=> ["SELECT DISTINCT ON(\"users\".\"id\") \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\""]
+```
+
+### except
+
+Creates a relation that is a combination of two relations with the EXCEPT operator.
+
+EXCEPT example
+
+```clojure
+(-> users-rel
+	(r/except admins-rel)
+	(r/get-select-query {}))
+
+=> ["SELECT \"users\".\"id\" AS \"id\", \"users\".\"is-admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM (SELECT \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\" EXCEPT SELECT \"users\".\"id\" AS \"id\", \"users\".\"is_admin\" AS \"is-admin\", \"users\".\"username\" AS \"username\" FROM \"users\" AS \"users\" WHERE \"users\".\"is_admin\" IS TRUE) AS \"users\""]
+```
+
+### extend-with-aggregate
```

Comment:
Can we add an intro for this? Does this basically just add a `count` to the select? What does the `extend` keyword mean?

Can we also specify how a sum would work - because sum needs a column name?

Reply:
I'll add an _Aggregate functions__ section and add examples.